### PR TITLE
Add reserved and recommended keywords for transport schemas

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

File: language/resource-api/README.md [hunk 1]
```diff
@@ -418,6 +418,26 @@ A transport connects providers to the remote target. It consists of the schema a
...
+
+The following keywords must not be used by the transport schema:
+
+* `name` - transports should use `uri` instead of name.
+* `path` - reserved as a uri part
+* `query` - reserved as a uri part
+* `run-on` - This is used by bolt to determine which target to proxy to. Transports should not rely on this key.
+* `remote-transport` - This is used to determine which transport to load. It should always be the transport class name "declassified".
+* `remote-*` Any key starting with `remote-` is reserved for future use.
+* `implementations`: reserved by boltThere are some reserved keywords that cannot be used as connection info attributes:
```
Comment (MikaelSmith) [inline language/resource-api/README.md]: Typo here. It also seems to be incomplete.

Comment (DavidS) [reply language/resource-api/README.md]: that was a copy&paste mistake - fixed. Thanks for noticing!

File: language/resource-api/README.md [hunk 2]
```diff
@@ -418,6 +418,24 @@ A transport connects providers to the remote target. It consists of the schema a
 
 Password attributes should also set `sensitive: true` to ensure that the data is handled securely. Attributes marked with this flag allow any UI based off this schema to make appropriate presentation choices. The value will be passed to the transport wrapped in a `Puppet::Pops::Types::PSensitiveType::Sensitive`. This will keep the value from being logged or saved inadvertently while it is being transmitted between components. To access the value within the Transport use the `unwrap` method. e.g. `connection_info[:password].unwrap`.
 
+The following keywords are encouraged within the transport schema:
+
+* `uri` - Use when you need to specify a specific URL to connect to. All of the following keys will be computed from the `uri` if possible. In the future more url parts may be computed from the URI as well.
+* `host` - Use to specify and IP or address to connect to.
```
Comment (clairecadman) [inline language/resource-api/README.md]: an*

File: language/resource-api/README.md [hunk 3]
```diff
@@ -418,6 +418,24 @@ A transport connects providers to the remote target. It consists of the schema a
...
+* `protocol` - Use to specify which protocol the transport should use for example `http`, `https`, `ssh` or `tcp`
+* `user` - The user the transport should connect as.
+* `port` - The port the transport should connect to.
+
+The following keywords must not be used by the transport schema:
+
+* `name` - transports should use `uri` instead of name.
+* `path` - reserved as a uri part
+* `query` - reserved as a uri part
+* `run-on` - This is used by bolt to determine which target to proxy to. Transports should not rely on this key.
```
Comment (clairecadman) [inline language/resource-api/README.md]: *Bolt

File: language/resource-api/README.md [hunk 4]
```diff
@@ -418,6 +418,24 @@ A transport connects providers to the remote target. It consists of the schema a
...
+* `remote-transport` - This is used to determine which transport to load. It should always be the transport class name "declassified".
+* `remote-*` Any key starting with `remote-` is reserved for future use.
+* `implementations`: reserved by bolt
```
Comment (clairecadman) [inline language/resource-api/README.md]: Bolt

**Comment by clairecadman** [comment]
@DavidS Do you think the key words should also be added to the main docs?

**Comment by DavidS** [comment]
Yes, the reserved keywords need to be listed everytime we describe how to write transport schemas.

**Comment by DavidS** [comment]
Thanks @clairecadman !

**Pull request merged**